src/frontend/src/optimizer/plan_node/plan_tree_node_v2.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use smallvec::SmallVec;
+
+pub trait PlanTreeNodeV2 {
+    type PlanRef;
+
+    fn inputs(&self) -> SmallVec<[Self::PlanRef; 2]>;
+    fn clone_with_inputs(&self, inputs: impl Iterator<Item = Self::PlanRef>) -> Self;
```

Comment:
I propose `set_inputs` what do you think

Reply:
just to make this traits more general :thinking: infact this trait is for those "planRef" instead of planNode